sopel/bot.py

```diff
@@ -53,7 +53,7 @@ def __init__(self, config, daemon=False):
         self.wantsrestart = False
         self._running_triggers = []
         self._running_triggers_lock = threading.Lock()
-        self._plugins = {}
+        self._plugins: Dict[str, Any] = {}
```

Comment:
That's what I mean by "basic": adding only the minimum things before going too far. In that case, this is also a reason for #2179 because when you declare type hint, you also have to import the thing you need, so you need to make sure you don't create cyclic-import.

In order to not change everything, I opted to use `Any` for now, not doing too much yet.